libavutil/hwcontext_cuda.c

```diff
@@ -387,28 +387,36 @@ static int cuda_device_create(AVHWDeviceContext *device_ctx,
     if (device)
         device_idx = strtol(device, NULL, 0);
 
-    if (cuda_device_init(device_ctx) < 0)
+    if (cuda_device_init(device_ctx) < 0) {
+        ret = AVERROR_UNKNOWN; //return non retriable error
```

Comment:
Whether error is retriable or not makes sense only on Broadcaster level when deciding if something is wrong with O+T, or the segment, so it shouldn't be mentioned here in a separate project. We might want to create an upstream PR for these changes later.